modules/react-native-wallet/android/src/main/java/com/wallet/WalletModule.kt

```diff
@@ -0,0 +1,304 @@
+          promise.resolve(
+            getCardStatusCode(token.tokenState)
+          )
+        } ?:  promise.resolve(CardStatus.NOT_FOUND_IN_WALLET.code)
+      }
+      .addOnFailureListener { e -> promise.reject("getCardStatus function failed", e) }
+      .addOnCanceledListener {
+        promise.reject(
+          "Reject",
+          "Card status retrieval canceled"
+        )
+      }
+  }
+
+  @ReactMethod
+  override fun getCardTokenStatus(tsp: String, tokenRefId: String, promise: Promise) {
+    if (!ensureTapAndPayClientInitialized()) {
+      promise.reject("Initialization error", "TapAndPay client initialization failed")
+      return
+    }
+
+    tapAndPayClient!!.getTokenStatus(getTokenServiceProvider(tsp), tokenRefId)
+      .addOnCompleteListener { task ->
+        if (!task.isSuccessful || task.result == null) {
+          promise.resolve(CardStatus.NOT_FOUND_IN_WALLET.code)
+          return@addOnCompleteListener
+        }
+        val token = task.result
+        token?.let {
+          promise.resolve(
+            getCardStatusCode(token.tokenState)
+          )
+        } ?:  promise.resolve(CardStatus.NOT_FOUND_IN_WALLET.code)
+      }
+      .addOnFailureListener { e -> promise.reject("getCardStatus function failed", e) }
+      .addOnCanceledListener {
+        promise.reject(
+          "Reject",
+          "Card status retrieval canceled"
+        )
+      }
+  }
+
+
+  @ReactMethod
+  override fun addCardToWallet(
+    data: ReadableMap, promise: Promise
+  ) {
+    if (!ensureTapAndPayClientInitialized()) return
+
+    try {
+      val cardData = data.toCardData() ?: return promise.reject("Reject: ", "Insufficient data")
+
+      val cardNetwork = getCardNetwork(cardData.network)
+      val tokenServiceProvider = getTokenServiceProvider(cardData.network)
+      if (cardNetwork == 1000 || tokenServiceProvider == 1000) {
+        return promise.reject("Reject: ", "Invalid card network")
+      }
+
+      val pushTokenizeRequest = PushTokenizeRequest.Builder()
+        .setOpaquePaymentCard(cardData.opaquePaymentCard.toByteArray(Charset.forName("UTF-8")))
+        .setNetwork(cardNetwork)
+        .setTokenServiceProvider(tokenServiceProvider)
+        .setDisplayName(cardData.cardHolderName)
+        .setLastDigits(cardData.lastDigits)
+        .setUserAddress(cardData.userAddress)
+        .build()
+
+      tapAndPayClient?.pushTokenize(
+        currentActivity!!, pushTokenizeRequest, REQUEST_CODE_PUSH_TOKENIZE
+      )
+
+      tapAndPayClient?.registerDataChangedListener {
+        Log.i("DUPA", "TEST")
+      }
+    } catch (e: java.lang.Exception) {
+      promise.reject(e)
+    }
+  }
+
+  private fun getWalletId(promise: Promise) {
+    if (!ensureTapAndPayClientInitialized()) {
+      promise.reject("Initialization error", "TapAndPay client initialization failed")
+      return
+    }
+    tapAndPayClient?.activeWalletId?.addOnCompleteListener { task ->
```

Comment:
```suggestion
    tapAndPayClient!!.activeWalletId?.addOnCompleteListener { task ->
```

Same. Not sure if `activeWalletId` is nullable by itself or due to `tapAndPayClient` being nullable, it's possible that the second `?` may be removed as well.